lib/packages.js

```diff
@@ -399,8 +409,24 @@ function _sortPackages(cache) {
     if(add) {
       sorted.push(pkg);
     } else {
+      const key = `${pkg.manifest.name}@${pkg.manifest.version}`;
+      cyclicDetector[key].count += 1;
+      const count = cyclicDetector[key].count;
+      maxCyclicCount = count >= maxCyclicCount ? count : maxCyclicCount;
       unsorted.push(pkg);
     }
+
+    if(maxCyclicCount >= cyclicDetectionThreshold) {
+      const badPackages = Object.keys(cyclicDetector).filter(key =>
+        (cyclicDetectionThreshold - cyclicDetector[key].count) <= 3)
+        .map(name => ({
+          name,
+          dependencies: cyclicDetector[name].dependencies
+        }));
+      logger.error('Found a cyclic dependency, look at these dependencies: ' +
+        `${JSON.stringify(badPackages, null, 2)}`);
+      process.exit(1);
```

Comment:
Would it be better to throw an exception with the error message and data vs log and exit?

Reply:
I think in this case it would definitely be a better option to throw an exception with the error message.